tensorflow/core/kernels/transpose_op.cc

```diff
@@ -133,7 +133,16 @@ void TransposeOp::Compute(OpKernelContext* ctx) {
   OP_REQUIRES(ctx, TensorShapeUtils::IsVector(perm.shape()),
               errors::InvalidArgument("perm must be rank 1, got shape ",
                                       perm.shape().DebugString()));
-
+  // Validation for negative values inside `perm`.
+  auto perm_vector = perm.vec<int>();
```

Comment:
The vector might not always be `int`. This causes some tests to fail

```
2024-03-20 05:50:59.400416: F tensorflow/core/framework/tensor.cc:844 Check failed: dtype() == expected_dtype (9 vs. 3) int32 expected, got int64
2024-03-20 05:50:59.400493: F tensorflow/core/framework/tensor.cc:844 Check failed: dtype() == expected_dtype (9 vs. 3) int32 expected, got int64
```

Reply:
The perm can accept int32 and int64. I can't see a template either to specify a generic dtype Tperm.  Any help would be appreciated.